lib/cloudregister/registerutils.py

```diff
@@ -575,7 +575,6 @@ def get_smt(cache_refreshed=None):
                     'Found alternate update server: '
                     '"%s"' % str((server.get_ipv4(), server.get_ipv6()))
                 )
-                replace_hosts_entry(current_smt, server)
```

Comment:
Logically removing this is correct, However. It misses the condition where for some reason there is no cache of the current update server but there is an entry in /etc/hosts. The "replace_hosts_entry()" call covered that condition. But obviously this creates other issues therefore the call should be replaced with a call to `clean_hosts_file()` and then we also need to update `/etc/hosts` with the new data, i.e. call `add_hosts_entry(server)`

Reply:
Makes sense, fixed. Perhaps it would make sense to refactor `replace_hosts_entry` to be:
```
def replace_hosts_entry(...):
    clean_hosts_file(...)
    add_hosts_entry(...)
```
?

Reply:
Yes makes sense, please go ahead and do the refactor. Thanks

Reply:
OK, done.